pkgs/development/python-modules/scancode-toolkit/default.nix

```diff
@@ -0,0 +1,112 @@
+{ lib
+, fetchPypi
+, buildPythonPackage
+, isPy3k
+, markupsafe
+, click
+, typecode
+, gemfileparser
+, pefile
+, fingerprints
+, spdx-tools
+, fasteners
+, pycryptodome
+, urlpy
+, dparse
+, jaraco_functools
+, pkginfo
+, debian-inspector
+, extractcode
+, ftfy
+, pyahocorasick
+, colorama
+, jsonstreams
+, packageurl-python
+, pymaven-patch
+, nltk
+, pygments
+, bitarray
+, jinja2
+, javaproperties
+, boolean-py
+, license-expression
+, extractcode-7z
+, extractcode-libarchive
+, typecode-libmagic
+}:
+let
+  bitarray' = bitarray.overridePythonAttrs ({ ... }: rec {
+    pname = "bitarray";
+    version = "0.9.3";
+    src = fetchPypi {
+      inherit pname version;
+      sha256 = "b2783d74247abb2669f581a001ecaa747e4b2ff9440762dbb499ed69b83e321e";
+    };
+  });
+in
+buildPythonPackage rec {
+  pname = "scancode-toolkit";
+  version = "21.3.31";
+  disabled = !isPy3k;
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "7e0301031a302dedbb4304a91249534f3d036f84a119170b8a9fe70bd57cff95";
+  };
+
+  preConfigure = ''
+    rm configure
+  '';
+
+  # * dparse2 is a "Temp fork for Python 2 support", but pdfminer requires
+  # Python 3, so it's "fine" to leave dparse2 unpackaged and use the "normal"
+  # version
+  # * ftfy was pinned for similar reasons (to support Python 2), but rather than
+  # packaging an older version, I figured it would be better to remove the
+  # erroneous (at least for our usage) version bound
+  preBuild = ''
```

Comment:
```suggestion
  postBuild = ''
```

Reply:
Hm, why should this go to `postBuild`? Finagling build-time dependencies won't work if I do it, well, after the build occurs (by which point the build will already have errored out).

Reply:
We usually use `postPatch`.